app/ui/pages/10_Settings.py

```diff
@@ -132,17 +134,28 @@ def task_listing(
 def save_tasks(container, wem, qhm, pm, user_limits):
     with container:
         with st.spinner("Saving changes to task settings."):
-            sql = f"""
-            call admin.run_as_app($$
-            begin
-                alter task TASKS.WAREHOUSE_EVENTS_MAINTENANCE {get_task_state(wem)};
-                alter task TASKS.QUERY_HISTORY_MAINTENANCE {get_task_state(qhm)};
-                alter task TASKS.SFUSER_MAINTENANCE {get_task_state(pm)};
-                alter task TASKS.USER_LIMITS_MAINTENANCE {get_task_state(user_limits)};
-            end;
-            $$);
-            """
-            connection.execute(sql)
+            with connection.Connection.get() as conn:
+                with snowpark_session(conn) as session:
+                    t = Task(task_name="WAREHOUSE_EVENTS_MAINTENANCE")
+                    if wem:
```

Comment:
Can this be factored to reduce duplication?

Especially to avoid redefining `t` to mean different things.

What you really have is a set of `{target_state: boolean, task_name: string}` objects that you're apply the same operation to.

Reply:
yup, I  was lazy. Fixed